pkg/log/zap/flags.go

```diff
@@ -158,3 +168,44 @@ func (v sampleValue) IsBoolFlag() bool {
 func (v sampleValue) Type() string {
 	return "sample"
 }
+
+type timeformatValue struct {
+	set         bool
+	timeEncoder zapcore.TimeEncoder
+	str         string
+}
+
+func (v *timeformatValue) Set(s string) error {
+	v.set = true
+
+	switch string(s) {
+	case "iso8601", "ISO8601":
+		v.timeEncoder = zapcore.ISO8601TimeEncoder
+	case "millis":
+		v.timeEncoder = zapcore.EpochMillisTimeEncoder
+	case "nanos":
+		v.timeEncoder = zapcore.EpochNanosTimeEncoder
+	default:
+		v.timeEncoder = zapcore.EpochTimeEncoder
+	}
+
+	return nil
+}
+
+func (v timeformatValue) String() string {
+	return v.str
+}
+
+func (v timeformatValue) IsBoolFlag() bool {
+	return false
+}
+
+func (v timeformatValue) Type() string {
+	return "string"
+}
+
+func withTimeFormat(te zapcore.TimeEncoder) encoderConfigFunc {
```

Comment:
Nit: actually, maybe rename this to `withTimeEncoder` so that it matches the type name of the parameter?